src/org/etools/j1939_84/controllers/part01/Part01Step08Controller.java

```diff
@@ -71,9 +72,18 @@ protected void run() throws Throwable {
         globalDM20s.forEach(this::save);
 
         // 6.1.8.2.a. Fail if minimum expected SPNs are not supported (in the aggregate response for the vehicle)
-        // per section A.4, Criteria for Monitor Performance Ratio Evaluation.
+        // per section A.4. When a numerator and denominator are provided as FFFF(h) and FFFF(h), the monitor identified
+        // in the label SPN shall be considered to be unsupported.
         Set<Integer> dm20Spns = globalDM20s.stream()
                                            .flatMap(dm20 -> dm20.getRatios().stream())
+                                           .peek(dm20 -> {
+                                               if (dm20.getNumerator() == 0xFF &&
```

Comment:
1. This needs to compare the numerator and denominator to 0xFFFF
2. This should be a filter.  The new logic is to consider the SPN to be unsupported if both the numerator and denominator are 0xFFFF.  That means the SPN is removed from the list.  Then the rest of the logic will take care of the failure. There's no need to report a new failure here.